Update to account for better @static

LilithHafner opened the pull request:
This is why I think @static works for `if`/`elseif`/`end`:
```
julia> if Sys.iswindows()
           $6
       elseif Sys.isapple()
           println("Apple.")
       else
           $9
       end
ERROR: syntax: "$" expression outside quote
Stacktrace:
 [1] top-level scope
   @ none:1

julia> @static if Sys.iswindows()
           $6
       elseif Sys.isapple()
           println("Apple.")
       else
           $9
       end
Apple.
```

LilithHafner committed:
Update to account for better @static

KristofferC commented:
But you didn't chain (nest) them there (see the example just below it)?

vtjnash commented:
It is true that `elseif` no longer should be in this list, but `if` still should

LilithHafner commented:
When I read "When chaining conditionals (including `if`/`elseif`/`end`)", I think of `if x; a; elseif z; b; else; c; end`, which could be expanded to two chained `if/else`s. It would be nice if the documentation made it clear that a single @static can cover an entire chain of `elseif`s.

LilithHafner committed:
Update handling-operating-system-variation.md

LilithHafner commented:
I've tried again with a revision. @KristofferC, I believe the term you used "nest" is better than "chain" because `if`...`elseif`...`else`...`end` is a chain but not a nest, so I switched that term.

I also added use of `elseif` to the existing example to clarify that it is supported.

@vtjnash I removed the list altogether. Do you think that creates any ambiguity? (e.g. someone thinking that nested `if`/`else`/`end`s would be supported)

vtjnash commented on doc/src/manual/handling-operating-system-variation.md:
```diff
@@ -27,13 +27,15 @@ Complex blocks:
 
-When chaining conditionals (including `if`/`elseif`/`end`), the `@static` must be repeated for
-each level (parentheses optional, but recommended for readability):
+When nesting conditionals, the `@static` must be repeated for each level 
```
```suggestion
When nesting conditionals, the `@static` must be repeated for each level
```

vtjnash committed:
Update doc/src/manual/handling-operating-system-variation.md
